### PR TITLE
{lang}[*] include zlib as a dependency in all Tcl easyconfigs (WIP)

### PR DESCRIPTION
**Pull request by boegel**
this syncs up existing easyconfigs like was done by @gjbex in #1340

the unit tests will check whether any conflicts were introduced at all (but I don't think there are)

@gjbex, @wpoely86: please review?

### CODE REVIEW

**Comment by wpoely86** [comment]
looks fine

but maybe a more permanent solution is to add it to the toolchain?

**Comment by boegel** [comment]
@wpoely86: that's a can of worms that we have to tackle carefully... but it's worth looking into, especailly for the common toolchains

**Comment by wpoely86** [comment]
No bigger can of worms then adding as a dep on all kinds of easyconfigs?

**Comment by boegel** [comment]
@wpoely86: I'm not sure... pushing zlib under GCC, or even binutils, is a lot lower in the stack, and affects *everything* being built with it

**Comment by wpoely86** [comment]
well, they now silently fall back to system zlib. It's gonna give a lot more of `no version information`. That can potentially break some build systems but we already have patched several pieces of software for that.

**Comment by boegel** [comment]
We can/should certainly look into this, but pulling zlib into the toolchain should also involve binutils, maybe even glibc (and more?)... Other common candidates are ncurses, libreadline, ...

**Comment by wpoely86** [comment]
Definitely not glibc. That's a real can of worms. And the same reasoning behind openssl also holds here (see GHOST).

Anyway, we should probably open a dedicated bug report for that?

**Comment by boegel** [comment]
@wpoely86: this one? https://github.com/hpcugent/easybuild-framework/issues/644

**Comment by wpoely86** [comment]
Yes, and now actually do it :smiley:

**Comment by boegel** [comment]
@wpoely86: Jenkins should be giving this a green light soon, last check? Good to go in?

File: easybuild/easyconfigs/z/zlib/zlib-1.2.5-ictce-5.5.0.eb [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+ not covered by any patents -- lossless data-compression library for use on virtually any 
+ computer hardware and operating system."""
+
+toolchain = {'name': 'ictce', 'version': '5.5.0'}
+toolchainopts = {'optarch': True, 'pic': True}
+
+sources = [SOURCELOWER_TAR_GZ]
+source_urls = [('http://sourceforge.net/projects/libpng/files/zlib/%(version)s', 'download')]
+
+preconfigopts = 'LDSHARED="$CC -shared"'
```
Comment (wpoely86) [inline easybuild/easyconfigs/z/zlib/zlib-1.2.5-ictce-5.5.0.eb]: drop this

**Comment by wpoely86** [comment]
looks fine

**Comment by boegel** [comment]
Test report by @boegel
**SUCCESS**
Build succeeded for 27 out of 27 (27 easyconfigs in this PR)
Linux SL 6.6, Intel(R) Xeon(R) CPU E5-2670 0 @ 2.60GHz, Python 2.6.6
See https://gist.github.com/54331d33d4920b70e9a8 for a full test report.

**Comment by boegel** [comment]
Updated to include recent Tcl easyconfigs, and to fix remarks.

Thanks for the review @wpoely86!

**Pull request merged**